### ISSUE TITLE
Update signature body format

### ISSUE THREAD

**Issue by webmaster128** [confio/weave#70]
Right now, the body to be signed is something like `<raw tx> | <chain ID> | <nonce>`, where `|` means binary concatenation and `<symbol>` a place holder.

This allows collisions when one body's raw tx ends with the same bytes as some other body's chain ID starts with. E.g.

```
<raw tx> | <chain ID> | <nonce>
===============================
aabbccdd|eeff|0011
aabbcc|ddeeff|0011
```

To avoid those kind of collisions, you either need field separators or have fixed field sizes. Separators are usually not practical when working on raw binary data.

Thus I propose the following format instead:

 ```
format version | chain ID length | chain ID            | nonce | bz
int32          | int32           | utf8 encoded string | int?  | raw data
```

**Comment by ethanfrey** [comment]
I would accept that proposal more or less with minor changes

format version | chain ID length | chain ID            | nonce | bz
4bytes          | int8           | utf8 encoded string | int64 (bigendian)  | raw data
the chain Id should be maximal ~32 bytes, so one byte for encoding the length is sufficient.
the nonce is a fixed 8 byte length, in BigEndian format. ChainId should be a subset of ascii, but we can define it as utf8 to be more permissive in parsing.

The version doesn't need just to be a counter 00000001, 00000002, ... but a unique identifier than can change between different versions. It also is a known magic number to inform how to interpret the following data.

We should use unique prefix(es) that are not simply a counter, to ensure minimal collision with other known formats and encodings. I propose the first version will be the prefix 00CAFE00 (no coffee), somewhat inspired by java class files CAFEBABE.

Updating this calculation will break the signing algorithm used in clients such as weave-js and should be accompanied with a version bump.

**Comment by ethanfrey** [comment]
Resolved by #73
